client/components/app.jsx

```diff
@@ -46,22 +81,36 @@ class App extends React.Component {
     super(props);
 
     this.state = {
+      ticker: null,
       companyName: null,
       latestPrice: null,
+      latestAfterHours: null,
       selectedPrice: null,
       refStartPrice: null,
       changeCaption: null,
-      defaultChangeCaption: null
+      defaultChangeCaption: null,
+      selectedGraph: '1D',
+      selectedCategory: null,
+      analystBuy: null,
+      platformOwners: null
     };
-
+    
     this.requestData = this.requestData.bind(this);
+    this.getYesterdayClose = this.getYesterdayClose.bind(this);
+    this.setTicker = this.setTicker.bind(this);
     this.setCompanyName = this.setCompanyName.bind(this);
     this.setSelectedPrice = this.setSelectedPrice.bind(this);
     this.setLatestPrice = this.setLatestPrice.bind(this);
+    this.setLatestAfterHours = this.setLatestAfterHours.bind(this);
     this.setRefStartPrice = this.setRefStartPrice.bind(this);
     this.setChangeCaption = this.setChangeCaption.bind(this);
     this.setDefaultChangeCaption = this.setDefaultChangeCaption.bind(this);
     this.handleMouseLeaveChart = this.handleMouseLeaveChart.bind(this);
+    this.handleOptionClick = this.handleOptionClick.bind(this);
```

Comment:
holy bindings, Batman! check out https://babeljs.io/docs/en/babel-plugin-proposal-class-properties if you want to avoid all the binding in the future